external-crates/move/crates/move-compiler/src/linters/unnecessary_unit.rs

```diff
@@ -1,47 +1,22 @@
 //! Detects an unnecessary unit expression in a block, sequence, if, or else.
 
 use crate::{
-    diag,
-    diagnostics::WarningFilters,
-    ice,
+    diag, ice,
     linters::StyleCodes,
-    shared::CompilationEnv,
     typing::{
         ast::{self as T, SequenceItem_, UnannotatedExp_},
-        visitor::{TypingVisitorConstructor, TypingVisitorContext},
+        visitor::simple_visitor,
     },
 };
 use move_ir_types::location::Loc;
 
-pub struct UnnecessaryUnit;
-
-pub struct Context<'a> {
-    env: &'a mut CompilationEnv,
-}
-
-impl TypingVisitorConstructor for UnnecessaryUnit {
-    type Context<'a> = Context<'a>;
-
-    fn context<'a>(env: &'a mut CompilationEnv, _program: &T::Program) -> Self::Context<'a> {
-        Context { env }
-    }
-}
-
-impl TypingVisitorContext for Context<'_> {
-    fn add_warning_filter_scope(&mut self, filter: WarningFilters) {
-        self.env.add_warning_filter_scope(filter)
-    }
-
-    fn pop_warning_filter_scope(&mut self) {
-        self.env.pop_warning_filter_scope()
-    }
-
+simple_visitor!(
```

Comment:
I have `simple_visitor!(` instead of something like `simple_visitor! {` to enable rustfmt